src/components/Catalog.vue

```diff
@@ -397,15 +404,17 @@ export default {
           } else {
             this.externalItemCount = items.features.length;
           }
+          const previousLink = items.links.find(link => link.rel === 'previous')
```

Comment:
These two properties should be computed properties.

Reply:
I'm not sure I'll be able to do that, the function that gather the external items doesn't expose then the `links` it receive. Or I could store them as `externalLinks` in data, and then extract linsk with computed.

Reply:
Yes, that's the issue with the current implementation. It adds yet another "data source", the `currentExternalItemsURL`. It should just re-use the existing mechanisms of loading a FeatureCollection, which is getEntity. This relates to https://github.com/radiantearth/stac-browser/pull/71#discussion_r590197530

I think I'd be fine with having your implementation in STAC Browser 1.0, but for 2.0 this needs a rewrite anyway.

Reply:
The best would be to have two distinct components, one for items given "as is" by the catalog and another one for component coming from external sources.
We could achieve that with a conditional component in the router, should I try to go down this way? (with another PR)

Reply:
Yes, that would make sense. But I'm planning to rewrite the whole Browser anyway from scratch in the next months for 2.0 as the current "architecture" is a mess. The "external" (it's not really external) is really just the way it is because no one yet cared to do it right and just tried to make it work somehow. In 2.0 this would just be handled internally in vuex, so I guess it's not worth it right now to spend more time fixing such issues. But I'd like to have your functionality in soon so that it can be used until we do the actual rewrite.

Reply:
I didn't mean to put all assets in the Collection. Just the requests itself would be "hidden" in vuex and you don't need to care whether something is an API response or a catalog. It would still lazy load data on request etc.